File: backend/src/main/java/com/fooddelivery/utils/MD5Util.java
package com.fooddelivery.utils;

import org.apache.commons.codec.digest.DigestUtils;

/**
 * MD5加密工具类
 */
public class MD5Util {

    /**
     * MD5加密
     */
    public static String encrypt(String text) {
        return DigestUtils.md5Hex(text);
    }

    /**
     * MD5加密（加盐）
     */
    public static String encrypt(String text, String salt) {
        return DigestUtils.md5Hex(text + salt);
    }

    /**
     * 验证密码
     */
    public static boolean verify(String text, String md5) {
        return encrypt(text).equalsIgnoreCase(md5);
    }

    /**
     * 验证密码（加盐）
     */
    public static boolean verify(String text, String salt, String md5) {
        return encrypt(text, salt).equalsIgnoreCase(md5);
    }
}

File: android-user/app/src/main/java/com/fooddelivery/user/utils/MD5Util.kt
package com.fooddelivery.user.utils

import java.security.MessageDigest

object MD5Util {
    fun encrypt(text: String): String {
        return try {
            val md = MessageDigest.getInstance("MD5")
            val digest = md.digest(text.toByteArray(Charsets.UTF_8))
            digest.joinToString("") { "%02x".format(it) }
        } catch (e: Exception) {
            text
        }
    }
}
